memtier_benchmark.cpp

```diff
@@ -844,7 +856,10 @@ run_stats run_benchmark(int run_id, benchmark_config* cfg, object_generator* obj
     // launch threads
     fprintf(stderr, "[RUN #%u] Launching threads now...\n", run_id);
     for (std::vector<cg_thread*>::iterator i = threads.begin(); i != threads.end(); i++) {
-        (*i)->start();
+        if (cfg->split)
+            (*i)->start(cfg->taskset.get_next_cpu());
+        else
+            (*i)->start();
```

Comment:
don't you need to provide a random core here (one from all available cores)?

Reply:
no, this method is called when running the threads without affinity